### PR TITLE
Add route

### PR DESCRIPTION
**Pull request by smuda**
Add a route to expose kubecost.

# How is this PR tested
There are three states for this PR to be tested in.

## No route
Install using `helm upgrade -i kubecost cost-analyzer -f ./values-openshift.yaml` and verify that there is no route.

## Route but no specific hostname
Install using `helm upgrade -i kubecost cost-analyzer -f ./values-openshift.yaml  --set route.enabled=true`

Verify there is a route:
`kubectl get route`

Verify that you can browse to the following address:
`kubectl get route kubecost-cost-analyzer-route -o jsonpath='{.status.ingress[0].host}'`

## Route with specific host
Install using `helm upgrade -i kubecost cost-analyzer -f ./values-openshift.yaml --set route.enabled=true --set route.host=hostname.apps.okd4.example.com`

Verify there is a route:
`kubectl get route`

Verify that you can browse to the address specifiked during installation.

### CODE REVIEW

File: values-openshift.yaml [hunk 1]
```diff
@@ -1,3 +1,6 @@
+route:
```
Comment (linhlam-kc) [inline values-openshift.yaml:1]: There are few keys are not defined. I suggest to add these in the values-openshift.yaml
```yaml
route:
  annotations: {}
  # Add custom dns URL for your route.
  # Example:  
  # host: kubecost.apps.okd4.example.com
 ```

**Review by linhlam-kc — CHANGES_REQUESTED** [review]
Thank you for your contribution ! I added few comments with suggested changes. Can you also share `How was this PR tested?` in the description of the PR.

File: values-openshift.yaml (hunk 1 above)
Comment (smuda) [reply values-openshift.yaml:1]: I've changed according to your suggestion.

File: cost-analyzer/templates/cost-analyzer-route.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  {{- with .Values.route.annotations }}
+  annotations:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+spec:
+  {{ if .Values.route.host }}
+  host: "{{ .Values.route.host }}""
+  {{- else }}
+  host: "kubecost-{{ .Release.Namespace }}.apps.okd4.example.com"
+  {{- end }}
```
[the remark this answers is not in the document]
Comment (smuda) [reply cost-analyzer/templates/cost-analyzer-route.yaml]: You're so right. I've changed to not include a default host but rather let OpenShift handle that.

**Comment by smuda** [comment]
I've also updated:

- the route does not anymore use a specific number as the port, but rather use the port name from the service.
- default to expose the route is false, to match earlier behavior.

**Review by linhlam-kc — APPROVED** [review]
Thank you again for your contribution John. The PR looks good to me now !

**Pull request merged**